codec/src/main/java/io/netty/handler/codec/compression/FastLz.java

```diff
@@ -246,55 +246,37 @@ static int compress(final byte[] input, final int inOffset, final int inLength,
             if (distance == 0) {
                 /* zero distance means a run */
                 //flzuint8 x = ip[-1];
-                byte x = input[inOffset + ip - 1];
+                byte x = input.getByte(inOffset + ip - 1);
                 while (ip < ipBound) {
-                    if (input[inOffset + ref++] != x) {
+                    if (input.getByte(inOffset + ref++) != x) {
                         break;
                     } else {
                         ip++;
                     }
                 }
             } else {
-                for (;;) {
-                    /* safe because the outer check against ip limit */
-                    if (input[inOffset + ref++] != input[inOffset + ip++]) {
-                        break;
-                    }
-                    if (input[inOffset + ref++] != input[inOffset + ip++]) {
-                        break;
-                    }
-                    if (input[inOffset + ref++] != input[inOffset + ip++]) {
-                        break;
-                    }
-                    if (input[inOffset + ref++] != input[inOffset + ip++]) {
-                        break;
-                    }
-                    if (input[inOffset + ref++] != input[inOffset + ip++]) {
-                        break;
-                    }
-                    if (input[inOffset + ref++] != input[inOffset + ip++]) {
-                        break;
-                    }
-                    if (input[inOffset + ref++] != input[inOffset + ip++]) {
-                        break;
-                    }
-                    if (input[inOffset + ref++] != input[inOffset + ip++]) {
+                /* safe because the outer check against ip limit */
+                boolean missMatch = false;
+                for (int i = 0; i < 8; i++) {
+                    if (input.getByte(inOffset + ref++) != input.getByte(inOffset + ip++)) {
+                        missMatch = true;
```

Comment:
Oh wow, it took me a while to figure out what the hey the original code was doing

Reply:
same here